printers/printerTree.php

```diff
@@ -42,15 +42,17 @@ function _orderTree($tree) {
         $this->_sorter->sort($tree->children);
```

Comment:
If I understand correctly (based on L150 and L153, `$this->children` may now contain elements that are either an array (that represent a page), or an instance of `NspagesTreeNodeNsNode` (whereas with the current code it can only be arrays).
I'm not sure I understand how the sorter is able to cope with those heterogeneous elements. I would be thrilled if you can help me understand how that works!

Reply:
No. Everything is an NspagesTreeNodeNsNode object. With this approach, there is no need to change anything in the sorting. If object has "f" node/data it is assigned to "self" property, if object is "d" node I create object and push it to children property (which is array). It has childrens that are objects of NspagesTreeNodeNsNode that have self property assigned with array of node data. 
With this approach we can build one beautiful tree without deep restriction with correct sorting (I tested it with 6 levels: a:b:c:d:e:f:start, then I added start page for "c", and move/rename it from "a:b:c:start" (a/b/c/start.txt) to "a:b:c" (a/b/c.txt)).

On the left nspages (with my patch) on right catlist plugin (without touching code). Sorting is that same:
![obraz](https://github.com/user-attachments/assets/5f783669-a715-45f0-9015-b463a7485aa9)

This patch also fix traversing my wiki using prev/next buttons in template. Earlier I had loops, because I had duplicates entries, I had proxmox in beginning of my tree level (before "ffmpeg") and at the end.